control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -503,7 +510,11 @@ func (r *Reconciler) finalizeNonExternalBrokerTopic(broker *eventing.Broker, sec
 	}
 	defer kafkaClusterAdminClient.Close()
 
-	topic, err := kafka.DeleteTopic(kafkaClusterAdminClient, kafka.BrokerTopic(TopicPrefix, broker))
+	topicName, err := r.KafkaFeatureFlags.ExecuteBrokersTopicTemplate(broker.ObjectMeta)
+	if err != nil {
+		return err
+	}
```

Comment:
We should take the topic name from the annotation to account for changes from when broker created to when broker is deleted.

Reply:
(same for channel reconciler)

Reply:
Good point!